src/onboarding/containers/LoginPage.tsx

```diff
@@ -9,30 +9,59 @@ import {
   InfluxDBCloudLogo,
   Typeface,
 } from '@influxdata/clockface'
+import {useHistory} from 'react-router-dom'
 import Notifications from 'src/shared/components/notifications/Notifications'
+import {client} from 'src/utils/api'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
 // Components
 import ErrorBoundary from 'src/shared/components/ErrorBoundary'
 import LoginPageContents from 'src/onboarding/containers/LoginPageContents'
 
-export const LoginPage: FC = () => (
-  <ErrorBoundary>
-    <AppWrapper>
-      <Notifications />
-      <FunnelPage
-        enableGraphic={true}
-        logo={<InfluxDBCloudLogo cloud={true} className="login-page--logo" />}
-      >
-        <Heading
-          element={HeadingElement.H1}
-          type={Typeface.Rubik}
-          weight={FontWeight.Regular}
-          className="cf-funnel-page--title"
+export const LoginPage: FC = () => {
+  const [hasValidSession, setHasValidSession] = useState(false)
+
+  const getSessionValidity = useCallback(async () => {
+    try {
+      await client.users.me()
```

Comment:
so this throws an exception if a non-200 status code is returned?

Reply:
yup. The API will return an error if the session doesn't exist. It's the same logic behind the current signin flow:

https://github.com/influxdata/ui/blob/master/src/Signin.tsx#L97